src/basic/path-util.h

```diff
@@ -85,6 +85,19 @@ static inline char* path_simplify(char *path) {
         return path_simplify_full(path, 0);
 }
 
+static inline int path_simplify_alloc(const char *path, char **ret) {
+        char *t = NULL;
+
+        if (path) {
+                t = strdup(path);
+                if (!t)
+                        return -ENOMEM;
+        }
+
+        *ASSERT_PTR(ret) = path_simplify(t);
```

Comment:
In this case I'd like to locate `assert(ret)` at the beginning of the function like the our traditional coding style, to make it easier to understand that the function requires non-NULL `ret`.